api/openapi/modelregistry.yml

```
@@ -0,0 +1,367 @@
+                RegisterdModel and must not be the same as any of the `id`s.
+    '/models/{name}/versions/{version}':
+        get:
+            responses:
+                '200':
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/VersionedModel'
+                    description: get `VersionedModel` for model name and version
+            summary: Get Version details
+        delete:
+            responses:
+                '200':
+                    description: Successfully archive a `VersionedModel`.
+            summary: Archive a VersionedModel
+            description: 'This does NOT delete any `VersionedModel`, but archives and deprecate it.'
+        patch:
+            requestBody:
+                description: 'The `uri`, `id`, `version` and other non-updatable fields will be ignored.'
+                content:
+                    application/json:
+                        schema:
+                            $ref: '#/components/schemas/VersionedModel'
+                required: true
+            responses:
+                '200':
+                    description: Successfully upated a `VersionedModel` in the updatable and provided fields.
+            description: |-
+                Updates a `VersionedModel` only for the updatable fields, such as label, tags, 
+                description.
+
+                Some fields will not be updated even if supplied, such as `model_uri`, `id`,
+                `version`.
+        parameters:
+            -
+                name: name
+                schema:
+                    type: string
+                in: path
+                required: true
+            -
+                name: version
+                schema:
+                    type: string
+                in: path
+                required: true
+    '/models/{name}':
+        get:
+            responses:
+                '200':
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/RegisteredModel'
+                    description: Retrieves the `RegisteredModel`.
+            summary: Return a given RegisteredModel
+        post:
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            $ref: '#/components/schemas/VersionedModel'
+            responses:
+                '200':
+                    description: Successfully registered a versionedmodel under a registered model.
+            summary: Register a VersionedModel under a given name
+            description: |-
+                TBD: this as a potential alternative or complementary of `/models` POST
+                where the model's name is provided as a Path param instead.
+        parameters:
+            -
+                name: name
```

Without this schema type, the generator complains during the auto-generation phase. Does this sound good to you?

```suggestion
                name: name
                schema:
                    type: string
```